infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/ReplicatingState.java

```diff
@@ -52,6 +52,12 @@ public LogReplicationRuntimeState processEvent(LogReplicationRuntimeEvent event)
                 log.debug("Connection lost to non-leader node {}", endpointDown);
                 // If a non-leader node loses connectivity, reconnect async and continue.
                 return null;
+            case REMOTE_LEADER_LOSS:
+                if (!fsm.getConnectedEndpoints().isEmpty() && fsm.getRemoteLeader().isPresent() && fsm.getRemoteLeader().get().equals(event.getEndpoint())) {
```

Comment:
What is actually needed here, is something like the following:

```
if(ism.getRemoteLeader().isPresent() && fsm.getRemoteLeader().get().equals(event.getEndpoint()) {
    fsm.resetRemoteLeaderEndpoint();
    return fsm.getConnectedEndpoints().isEmpty() ? fsm.getStates().get(LogReplicationRuntimeStateType.WAITING_FOR_CONNECTIVITY) :
                            fsm.getStates().get(LogReplicationRuntimeStateType.VERIFYING_REMOTE_LEADER);
```

Reply:
Ok I get it now